### PR TITLE
JBAI-13933 Extend Prompt builder tests

### PR DESCRIPTION
**Pull request by aozherelyeva**
Added new tests for the Prompt builder (the necessity was previously discussed when there was a bug of not applying the temperature to the prompt).

---

#### Type of the change
- [x] New feature
- [ ] Bug fix
- [ ] Documentation fix

#### Checklist for all pull requests
- [x] The pull request has a description of the proposed change
- [x] I read the [Contributing Guidelines](https://github.com/JetBrains/koog/blob/main/CONTRIBUTING.md) before opening the pull request
- [x] The pull request uses **`develop`** as the base branch
- [x] Tests for the changes have been added
- [x] All new and existing tests passed

##### Additional steps for pull requests adding a new feature
- [x] An issue describing the proposed change exists
- [ ] The pull request includes a link to the issue
- [ ] The change was discussed and approved in the issue
- [ ] Docs have been added / updated

### CODE REVIEW

**Review by Ololoshechkin — APPROVED** [review]

File: prompt/prompt-model/src/commonTest/kotlin/ai/koog/prompt/PromptTest.kt [hunk 1]
```diff
@@ -1,25 +1,35 @@
...
+import ai.koog.prompt.params.LLMParams
 import kotlinx.serialization.encodeToString
 import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.buildJsonObject
+import kotlinx.serialization.json.put
 import kotlin.test.Test
 import kotlin.test.assertEquals
 import kotlin.test.assertTrue
 
 class PromptTest {
```
Comment (skarpovdev) [inline prompt/prompt-model/src/commonTest/kotlin/ai/koog/prompt/PromptTest.kt]: Let's also add tests for negative scenarios like passing empty/invalid/null parameters/messages.
It's worth checking if any of those functions throw any exceptions in certain conditions and check these cases in tests.

File: prompt/prompt-model/src/commonTest/kotlin/ai/koog/prompt/PromptTest.kt [hunk 2]
```diff
@@ -54,4 +64,110 @@ class PromptTest {
...
+
+        assertEquals(3, updatedPrompt.messages.size)
+        assertEquals(systemMessage, updatedPrompt.messages[0].content)
+        assertEquals(userMessage, updatedPrompt.messages[1].content)
+        assertEquals(assistantMessage, updatedPrompt.messages[2].content)
+    }
+
+    @Test
+    fun testUpdatePromptWithNewParams() {
+        val originalPrompt = Prompt.build("test") {
```
Comment (skarpovdev) [inline prompt/prompt-model/src/commonTest/kotlin/ai/koog/prompt/PromptTest.kt]: Can it be a property since it's repeated multiple times to reduce duplication?

**Comment by KarpovSS** [comment]
I know `testSerialization` wasn't modified in this PR, but it seems this test could be much more extensive to cover all message types as well as parameters. Feel free to ignore this comment if you don't think it should be done in this PR though.

File: prompt/prompt-model/src/commonTest/kotlin/ai/koog/prompt/PromptTest.kt (hunk 1 above)
Comment (zarechneva-a) [reply prompt/prompt-model/src/commonTest/kotlin/ai/koog/prompt/PromptTest.kt]: Valid point, thank you!

**Review by KarpovSS — APPROVED** [review]
